lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -2117,11 +2117,11 @@ def _download_and_push_promtail_to_workload(self, promtail_info: dict) -> None:
         """
         # Check for Juju proxy variables and fall back to standard ones if not set
         proxies: Optional[Dict[str, str]] = {}
-        if proxies and os.environ.get("JUJU_CHARM_HTTP_PROXY"):
+        if isinstance(proxies, dict) and os.environ.get("JUJU_CHARM_HTTP_PROXY"):
             proxies.update({"http": os.environ["JUJU_CHARM_HTTP_PROXY"]})
-        if proxies and os.environ.get("JUJU_CHARM_HTTPS_PROXY"):
+        if isinstance(proxies, dict) and os.environ.get("JUJU_CHARM_HTTPS_PROXY"):
             proxies.update({"https": os.environ["JUJU_CHARM_HTTPS_PROXY"]})
-        if proxies and os.environ.get("JUJU_CHARM_NO_PROXY"):
+        if isinstance(proxies, dict) and os.environ.get("JUJU_CHARM_NO_PROXY"):
             proxies.update({"no_proxy": os.environ["JUJU_CHARM_NO_PROXY"]})
         else:
             proxies = None
```

Comment:
This means that if you set JUJU_CHARM_HTTP_PROXY and JUJU_CHARM_HTTPS_PROXY but not JUJU_CHARM_NO_PROXY `proxies` will be `None`. Is that correct?

Reply:
No this is not correct. Thanks for pointing that out. We want `proxies` to be `None` only if nothing was set (I can't think of a sound case where JUJU_HTTP_PROXY is set on one side and HTTPS_PROXY on the other)

Reply:
Would it work to have
```python
            proxies = {
                "https_proxy": os.environ.get("JUJU_CHARM_HTTPS_PROXY", ""),
                "http_proxy": os.environ.get("JUJU_CHARM_HTTP_PROXY", ""),
                "no_proxy": os.environ.get("JUJU_CHARM_NO_PROXY", ""),
            }
```

If you confirmed that an empty `""` value breaks things, then perhaps add:
```python
            proxies = {
                "https_proxy": os.environ.get("JUJU_CHARM_HTTPS_PROXY"),
                "http_proxy": os.environ.get("JUJU_CHARM_HTTP_PROXY"),
                "no_proxy": os.environ.get("JUJU_CHARM_NO_PROXY"),
            }
            proxies = {k: v for k, v in proxies.items() if v is not None} or None
```

Reply:
The second would logically work (it needs to be `None` if we don't want to mask the usual env variables).  
But mypy wants `Optional[Dict[str, str]]` as an argument for the proxyHandler.  
So I could try to do this:
```
        proxies: Optional[Dict[str, str]] = {
            "https_proxy": os.environ.get("JUJU_CHARM_HTTPS_PROXY"),
            "http_proxy": os.environ.get("JUJU_CHARM_HTTP_PROXY"),
            "no_proxy": os.environ.get("JUJU_CHARM_NO_PROXY"),
        }
        if isinstance(proxies, dict):
            proxies = {k: v for k, v in proxies.items() if v is not None} or None

        proxy_handler = request.ProxyHandler(proxies)
```

But it doesn't work because here `proxies` is in fact `Optional[Dict[str, Optional[str]]]`. So mypy complains again (did I mentioned that I love mypy ?).  
At that point I find my proposal easier to read and understand.

Reply:
Just after writing that, I understood that I could use the idea from your first version to write it in a way that mypy accepts (using an empty string instead of None)